apps/client/src/features/app-settings/panel/sources-panel/import-map/importMapUtils.ts

```diff
@@ -82,3 +83,19 @@ export function getPersistedOptions(): NamedImportMap {
     return namedImportMap;
   }
 }
+
+export function addExistingCustomFields(importMap: NamedImportMap, customFields: CustomFields): NamedImportMap {
+  const custom = importMap.custom;
+  Object.entries(customFields).forEach(([key, { label }]) => {
+    if (
+      !custom.some((ic) => {
+        return ic.ontimeName === key;
+      })
+    ) {
+      // const ontimeName = key as unknown as string;
+      custom.push({ ontimeName: key, importName: label });
+    }
+  });
+
+  return importMap;
+}
```

Comment:
question: could you please add some comments as to what this does?
We prefer explicit in favour of terser logic. Terser is a job for the builder